wallet/wallet.h

```diff
@@ -38,6 +39,14 @@ enum wallet_output_type {
 	htlc_recv = 4
 };
 
+/* A database backed shachain struct. The datastructure is
+ * writethrough, reads are performed from an in-memory version, all
+ * writes are passed through to the DB. */
+struct wallet_shachain {
+	u64 id;
+	struct shachain chain;
+};
+
```

Comment:
This stuff, of course, does not belong in the wallet.  But we can hoist the db code up later...